src/app/screens/Options/TestConnection/index.tsx

```diff
@@ -4,34 +4,39 @@ import Loading from "@components/Loading";
 import { useEffect, useState } from "react";
 import { useTranslation } from "react-i18next";
 import { useNavigate } from "react-router-dom";
-import { useAccount } from "~/app/context/AccountContext";
+import useSWR from "swr";
+import {
+  buildAccountInfo,
+  getAccountInfoKey,
+  useAccount,
+} from "~/app/context/AccountContext";
 import { useAccounts } from "~/app/context/AccountsContext";
 import { useSettings } from "~/app/context/SettingsContext";
-import api from "~/common/lib/api";
+import api, { AccountInfoRes, getAccountInfo } from "~/common/lib/api";
 import msg from "~/common/lib/msg";
-import type { AccountInfo } from "~/types";
+import { AccountInfo } from "~/types";
 
 export default function TestConnection() {
   const { getFormattedInCurrency } = useSettings();
-  const auth = useAccount();
   const { getAccounts } = useAccounts();
+  const { setAccountId } = useAccount();
 
-  const [accountInfo, setAccountInfo] = useState<{
-    alias: AccountInfo["alias"];
-    name: AccountInfo["name"];
-    balance: AccountInfo["balance"];
-    currency: AccountInfo["currency"];
-  }>();
+  const [accountInfo, setAccountInfo] = useState<Partial<AccountInfo>>();
+  const [currentAccountId, setCurrentAccountId] = useState<string>("");
   const [errorMessage, setErrorMessage] = useState("");
   const [loading, setLoading] = useState(false);
+  const { data: accountInfoRes } = useSWR<AccountInfoRes>(
```

Comment:
The TestConnection is actually a one time action after adding a new account. Does it even make sense to apply `useSWR` in this case? Wouldn't it even make more sense to just apply the fetcher function directly? 

`export const getAccountInfo = (): Promise<AccountInfoRes> =>
  msg.request<AccountInfoRes>("accountInfo")`

Reply:
it will save the account in the SWR in memory cache for the next use (otherwise in this case it would have to fetch the account info twice), so it is nice to use SWR in my opinion